dockstore-webservice/src/main/java/io/dockstore/webservice/resources/MetadataResource.java

```diff
@@ -414,4 +424,50 @@ public Config getConfig() {
         }
     }
 
+    @GET
+    @Timed
+    @Path("cli-info")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "Get Dockstore CLI information.", description = "Get Dockstore CLI information. NO authentication")
+    @ApiOperation(value = "Get Dockstore CLI information.", notes = "Get Dockstore CLI information. NO authentication")
+    public CLIInfo getCliVersion() {
+        GHRelease ghRelease;
+        PagedIterable<GHRelease> allReleases;
+        CLIInfo cliInfo = new CLIInfo();
+        try {
+            GitHub github = new GitHubBuilder().build();
+            GHRepository repository = github.getRepository("dockstore/dockstore-cli");
+            ghRelease = repository.getLatestRelease();
+            allReleases = repository.listReleases();
+        } catch (IOException e) {
+            LOG.info("Could not get CLI releases information from GitHub. ", e);
+            throw new CustomWebApplicationException("Could not get CLI  releases information info from GitHub",
+                    HttpStatus.SC_SERVICE_UNAVAILABLE);
+        }
+        String cliLatestVersion = ghRelease.getName();
+        cliInfo.setCliLatestDockstoreScriptDownloadUrl(DOCKSTORE_CLI_RELEASES_URL + "/download/" + cliLatestVersion + "/dockstore");
+        cliInfo.setCliLatestVersion(cliLatestVersion);
+        // Get the latest unstable version
+        // If there is no unstable version return an empty string as the latest unstable version
+        String cliLatestUnstableVersion = getLatestUnstableVersion(allReleases).orElse("");
```

Comment:
Does the CLI currently allow the unstable version to be older than stable version (I don't know)? The idea of an unstable version in my mind would be to get the latest and greatest, at the risk of it being untested, as opposed to downgrading to an older untested version.

So I think you should return an unstable version only if it's greater than the stable version. But if the CLI script currently behaves like this already, then that could be an improvement for a future release.

Reply:
Yes the CLI currently allows the unstable version to be older than stable version.

Reply:
Now returns a prerelease only if the prerelease is greater than the latest release